File: iteration1/src/CSE3063F22P1_GRP9/Course.java
package CSE3063F22P1_GRP9;

import java.util.List;

public class Course {
	private String ID;
	private String name;
	private Course prerequisite;
	private String prerequisiteId;
	private int quota;
	private int credit;
	private String semester;
	
	public String getID() {
		return ID;
	}
	public void setID(String iD) {
		ID = iD;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public Course getPrerequisite() {
		return prerequisite;
	}
	public void setPrerequisite(Course prerequisite) {
		this.prerequisite = prerequisite;
	}
	public String getPrerequisiteId() {
		return prerequisiteId;
	}
	public void setPrerequisiteId(String prerequisiteId) {
		this.prerequisiteId = prerequisiteId;
	}
	public int getQuota() {
		return quota;
	}
	public void setQuota(int quota) {
		this.quota = quota;
	}
	
	public int getCredit() {
		return credit;
	}
	public void setCredit(int credit) {
		this.credit = credit;
	}
	
	public String getSemester() {
		return semester;
	}
	public void setSemester(String semester) {
		this.semester = semester;
	}
	
}

File: iteration3/src/Course.py

from CourseStatistics import CourseStatistics

class Course:

    def __init__(self):
        self._courseStatistics = CourseStatistics(self)
        self._ID = 0 
        self._name = ''
        self._prerequisite = None
        self._prerequisiteId = None
        self._quota = 0
        self._credit = 0
        self._semsester = ''
        self._scheduleList = []
    
    def checkCollision(self,course):
        for schedule in self._scheduleList:
            for schedule2 in course.getScheduleList():
                if(schedule.checkCollision(schedule2)):
                    return True #Courses collides
        return False #No collision
    
    def getID(self):
        return self._ID

    
    def setID(self, id):
        self._ID = id

    
    def getName(self):
        return self._name

    
    def setName(self, name):
        self._name = name

   
    def getPrerequisite(self):
        return self._prerequisite

   
    def setPrerequisite(self, prerequisite):
        self._prerequisite = prerequisite

    
    def getPrerequisiteId(self):
        return self._prerequisiteId

    
    def setPrerequisiteId(self, prerequisiteId):
        self._prerequisiteId = prerequisiteId

    def getQuota(self):
        return self._quota

   
    def setQuota(self, quota):
        self._quota = quota

   
    def getCredit(self):
        return self._credit

   
    def setCredit(self, credit):
        self._credit = credit

   
    def getSemester(self):
        return self._semester

   
    def setSemester(self, semester):
        self._semester = semester

    
    def getCourseStatistics(self):
        return self._courseStatistics

    def getScheduleList(self):
        return self._scheduleList
    
    def addSchedule(self,schedule):
        self._scheduleList.append(schedule)